spi/src/main/java/org/eclipse/dataspaceconnector/spi/Result.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright (c) 2021 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.spi;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+
+public class Result<T> {
```

Comment:
One thing to think about: there are cases when we need to distinguish the type of failure, for example, is it fatal or can it be retried.

Reply:
I see, like it's handled in `HttpFunctionDataFlowController`. 
I think we can choose between two naming conventions:
```
Result.success
Result.failure
Result.fatalFailure
```

or
```
Result.ok
Result.error
Result.fatalError
```

Reply:
Looks good, I don't have strong preference either way, but maybe "success"?